addons/udes_warehouse_classification/models/stock_picking.py

```diff
@@ -6,16 +6,24 @@ class StockPicking(models.Model):
     _inherit = "stock.picking"
 
     def _get_classification_messages_for_product_picking(self):
-        """Return the product with classification messages appropriate for a picking"""
+        """
+        Return all products with classification messages appropriate to warn users. It is getting
+        all products, not only the products of the picking because it may be some products with
+        classifications that are not inside the picking but we still need to warn users.
+        """
+        Product = self.env["product.product"]
+        self.ensure_one()
         product_classifications = super()._get_classification_messages_for_product_picking()
-        products = self.mapped("move_lines.product_id")
+        products = Product.search([("u_has_warehouse_classifications", "=", True)])
```

Comment:
Q: why do we do this on every product in the system? I think if a warehouse user has cherries, and bananas in his, he should only get warning messages about those products not the ones not in his picking. Maybe a question for a PO.

Reply:
To me it seems like we only want to do the check on the expected and unexpected items that has been scanned in, not all the products in a specific picking_type. Not sure how that would be done though.

Reply:
Yes but unexpected items are products which are not on moves that why are unexpected. 
Check the technical notes on the story if that helps

Reply:
Yeah I understand now